data-prepper-plugins/mongodb/src/main/java/org/opensearch/dataprepper/plugins/mongo/converter/RecordConverter.java

```diff
@@ -0,0 +1,144 @@
+     * @param eventCreationTimeMillis Creation timestamp of the event
+     * @param eventName               Event name
+     * @throws Exception Exception if failed to write to buffer.
+     */
+    public void addToBuffer(final AcknowledgementSet acknowledgementSet,
+                            final Map<String, Object> data,
+                            final Map<String, Object> keys,
+                            final long eventCreationTimeMillis,
+                            final long eventVersionNumber,
+                            final String eventName) throws Exception {
+        final Event event = JacksonEvent.builder()
+                .withEventType(getEventType())
+                .withData(data)
+                .build();
+
+        // Only set external origination time for stream events, not export
+        if (eventName != null) {
+            final Instant externalOriginationTime = Instant.ofEpochMilli(eventCreationTimeMillis);
+            event.getEventHandle().setExternalOriginationTime(externalOriginationTime);
+            event.getMetadata().setExternalOriginationTime(externalOriginationTime);
+        }
+        final EventMetadata eventMetadata = event.getMetadata();
+
+        eventMetadata.setAttribute(MetadataKeyAttributes.EVENT_COLLECTION_METADATA_ATTRIBUTE, collectionConfig.getCollection());
+        eventMetadata.setAttribute(MetadataKeyAttributes.EVENT_TIMESTAMP_METADATA_ATTRIBUTE, eventCreationTimeMillis);
+        eventMetadata.setAttribute(MetadataKeyAttributes.MONGODB_STREAM_EVENT_NAME_METADATA_ATTRIBUTE, eventName);
+        eventMetadata.setAttribute(MetadataKeyAttributes.EVENT_NAME_BULK_ACTION_METADATA_ATTRIBUTE, mapStreamEventNameToBulkAction(eventName));
+        eventMetadata.setAttribute(MetadataKeyAttributes.EVENT_VERSION_FROM_TIMESTAMP, eventVersionNumber);
+
+        final String partitionKey = getAttributeValue(keys, MetadataKeyAttributes.MONGODB_PRIMARY_KEY_ATTRIBUTE_NAME);
+        eventMetadata.setAttribute(MetadataKeyAttributes.PARTITION_KEY_METADATA_ATTRIBUTE, partitionKey);
+        eventMetadata.setAttribute(MetadataKeyAttributes.PRIMARY_KEY_DOCUMENT_ID_METADATA_ATTRIBUTE, partitionKey);
+
+        if (acknowledgementSet != null) {
+            acknowledgementSet.add(event);
+        }
+        bufferAccumulator.add(new Record<>(event));
+    }
+
+    public void addToBuffer(final AcknowledgementSet acknowledgementSet,
+                            final Map<String, Object> data,
+                            final long timestamp,
+                            final long eventVersionNumber) throws Exception {
+        addToBuffer(acknowledgementSet, data, data, timestamp, eventVersionNumber, null);
+    }
+
+    private String mapStreamEventNameToBulkAction(final String streamEventName) {
+        if (streamEventName == null) {
+            return DEFAULT_ACTION;
+        }
+
+        switch (streamEventName) {
+            case "INSERT":
```

Comment:
Will these event names be the same for this source? These aren't unique to DDB streams?

Reply:
https://www.mongodb.com/docs/manual/reference/change-events/

I haven't integrated streams end to end for mongoDb/DocumentDB. Will update if following PRs this week. Mongo also follows the same event types - insert, modify and delete. In additional there are other event type that needs to be handled.